src/DssSpell.sol

```diff
@@ -16,25 +16,108 @@
+    bytes                  public sig;
+    bool                   public done;
+    bytes32      immutable public tag;
+    address      immutable public action;
+    uint256      immutable public expiration;
+    PauseLike    immutable public pause;
+
+    uint256 constant public MIN_ETA = 1726574400; // 2024-09-17T12:00:00Z
+
+    // Provides a descriptive tag for bot consumption
+    // This should be modified weekly to provide a summary of the actions
+    // Hash: seth keccak -- "$(wget https://<executive-vote-canonical-post> -q -O - 2>/dev/null)"
+    function description() external view returns (string memory) {
+        return SpellActionLike(action).description();
+    }
+
+    function officeHours() external view returns (bool) {
+        return SpellActionLike(action).officeHours();
+    }
+
+    function nextCastTime() external view returns (uint256 castTime) {
+        return SpellActionLike(action).nextCastTime(eta);
+    }
+
+    // @param _description  A string description of the spell
+    // @param _expiration   The timestamp this spell will expire. (Ex. block.timestamp + 30 days)
+    // @param _spellAction  The address of the spell action
+    constructor(uint256 _expiration, address _spellAction) {
+        pause       = PauseLike(chainlog.getAddress("MCD_PAUSE"));
+        expiration  = _expiration;
+        action      = _spellAction;
+
+        sig = abi.encodeWithSignature("execute()");
+        bytes32 _tag;                    // Required for assembly access
+        address _action = _spellAction;  // Required for assembly access
+        assembly { _tag := extcodehash(_action) }
+        tag = _tag;
+    }
+
+    function schedule() public {
+        require(block.timestamp <= expiration, "This contract has expired");
+        require(eta == 0, "This spell has already been scheduled");
+        // Even if the spell gathers enough support too fast, we do not allow it to be executed before MIN_ETA.
+        eta = _max(block.timestamp + PauseLike(pause).delay(), MIN_ETA);
+        pause.plot(action, tag, sig, eta);
+    }
+
+    function cast() public {
+        require(!done, "spell-already-cast");
+        done = true;
+        pause.exec(action, tag, sig, eta);
+    }
+
+    function _max(uint256 a, uint256 b) internal pure returns (uint256) {
+        return a > b ? a : b;
+    }
 }
 
 contract DssSpellAction is DssAction {
     // Provides a descriptive tag for bot consumption
     // This should be modified weekly to provide a summary of the actions
     // Hash: cast keccak -- "$(wget 'https://raw.githubusercontent.com/makerdao/community/3c1ea8b373f3fc30885619ddcc8ee7aa2be0030a/governance/votes/Executive%20vote%20-%20September%205%2C%202024.md' -q -O - 2>/dev/null)"
     string public constant override description =
-        "2024-09-05 MakerDAO Executive Spell | Hash: 0xf387f3329537d6495861cd05b824ef1e5d6d2c15e5e8f0ae68e0928d88b01f37";
+        "2024-09-17 MakerDAO Executive Spell | Hash: TODO";
```

Comment:
I think this suppose to be `2024-09-13` based on the exec sheet name and even the title of this PR

Reply:
Yeah, I mixed it with the execution day. I fixed locally, will push soon.